dozer/cogs/tba.py

```diff
@@ -37,100 +41,271 @@ def __init__(self, bot):
+        e.description = events
+        await ctx.send(embed=e)
+
+    eventsfor.example_usage = """
+    `{prefix}tba eventsfor 1533` - show the currently registered events for team 1533, Triple Strange
+    """
+
+    @tba.command()
+    @bot_has_permissions(embed_links=True)
+    async def media(self, ctx, team_num: int, year: int = None):
+        """Get media of a team for a given year. Defaults to current year."""
+        if year is None:
+            year = datetime.datetime.today().year
+        try:
+            team_media = await self.session.team_media(team_num, year)
+
+            pages = []
+            base = f"FRC Team {team_num} {year} Media: "
+            for media in team_media:
+
+                if media.type == "youtube":
+                    pages.append(f"**{base} YouTube** \nhttps://youtu.be/{media.foreign_key}")
+                    continue
+                else:
+                    name, url, img_url = {
+                        "cdphotothread": (
+                            "Chief Delphi",
+                            "https://www.chiefdelphi.com/media/photos/{foreign_key}",
+                            "https://www.chiefdelphi.com/media/img/{image_partial}"
+                        ),
+                        "imgur": (
+                            "Imgur",
+                            "https://imgur.com/{foreign_key}",
+                            "https://i.imgur.com/{foreign_key}.png"
+                        ),
+                        "instagram-image": (
+                            "instagram",
+                            "https://www.instagram.com/p/{foreign_key}",
+                            "https://www.instagram.com/p/{foreign_key}/media"
+                        ),
+                        "grabcad": (
+                            "GrabCAD",
+                            "https://grabcad.com/library/{foreign_key}",
+                            "{model_image}"
+                        )
+                    }.get(media.type, (None, None, None))
+                    media.details['foreign_key'] = media.foreign_key
+                    page = discord.Embed(title=base + name, url=url.format(**media.details))
+                    page.set_image(url=img_url.format(**media.details))
+                    pages.append(page)
+
+            if len(pages):
+                await paginate(ctx, pages)
+            else:
+                await ctx.send(f"No media for team {team_num} found in {year}!")
+
+        except aiotba.http.AioTBAError:
+            raise BadArgument("Couldn't find data for team {}".format(team_num))
+
+    media.example_usage = """
+    `{prefix}tba media 971 2016` - show available media from team 971 Spartan Robotics in 2016
+    """
+
+    @tba.command()
+    @bot_has_permissions(embed_links=True)
+    async def awards(self, ctx, team_num: int, year: int = None):
+        """Gets a list of awards the specified team has won during a year. """
+        with ctx.typing():
```

Comment:
`ctx.typing()` should be used in an async-with so typing starts immediately, rather than when this task next yields.